src/forms/UncontrolledForm.tsx

```diff
@@ -0,0 +1,173 @@
+import { useRef, useState } from 'react'
+import { nanoid } from 'nanoid'
+import { formSchema } from './schema'
+import { getPasswordStrength } from './password'
+import { isValidImage, toBase64 } from './image'
+import CountryAutocomplete from './CountryAutocomplete'
+import { useAppDispatch } from '../store/hooks'
+import { upsertEntry } from '../store/formsSlice'
+
+type Props = { onDone: () => void; compact?: boolean }
+
+export default function UncontrolledForm({ onDone, compact = false }: Props) {
+  const f = useRef<HTMLFormElement | null>(null)
+  const [errors, setErrors] = useState<Record<string, string>>({})
+  const [country, setCountry] = useState('')
+  const [pwd, setPwd] = useState('')
+  const [gender, setGender] = useState<'male' | 'female' | 'other'>('other')
+  const [locked, setLocked] = useState(false)
+  const dispatch = useAppDispatch()
+
+  const onSubmit = async (e: React.FormEvent) => {
+    e.preventDefault()
+    const fd = new FormData(f.current!)
+    const image = fd.get('image') as File
+    let imageBase64: string | undefined
+    if (image && image.size > 0) {
```

Comment:
And in other such places
```suggestion
    let imageBase64: string | undefined
    
    if (image && image.size > 0) {
```